akka-typed/src/main/scala/akka/typed/cluster/ddata/scaladsl/Replicator.scala

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.typed.cluster.ddata.scaladsl
+
+import akka.actor.NoSerializationVerificationNeeded
+
+import akka.cluster.{ ddata ⇒ dd }
+import akka.cluster.ddata.Key
+import akka.cluster.ddata.ReplicatedData
+import akka.typed.ActorRef
+import akka.typed.Behavior
+import akka.typed.cluster.ddata.internal.ReplicatorBehavior
+
+object Replicator {
+
+  def behavior(settings: ReplicatorSettings): Behavior[Command[_]] =
+    ReplicatorBehavior.behavior(settings)
+
+  type ReadConsistency = dd.Replicator.ReadConsistency
+  val ReadLocal = dd.Replicator.ReadLocal
+  type ReadFrom = dd.Replicator.ReadFrom
+  type ReadMajority = dd.Replicator.ReadMajority
+  type ReadAll = dd.Replicator.ReadAll
+
+  type WriteConsistency = dd.Replicator.WriteConsistency
+  val WriteLocal = dd.Replicator.WriteLocal
+  type WriteTo = dd.Replicator.WriteTo
+  type WriteMajority = dd.Replicator.WriteMajority
+  type WriteAll = dd.Replicator.WriteAll
+
+  sealed trait Command[A <: ReplicatedData] {
+    def key: Key[A]
+  }
+
+  final case class Get[A <: ReplicatedData](key: Key[A], consistency: ReadConsistency, request: Option[Any] = None)(val replyTo: ActorRef[GetResponse[A]])
```

Comment:
the reason I placed `replyTo` in a second parameter list was that then it's nice with `ask`, but on the other hand `ask` is not the primary way you would interact with the Replicator so perhaps that is not worth it. It becomes rather ugly for the `Update` message that already has one extra param list.

one could always use `ref => Get(...)` for ask